roles/kafka_broker/tasks/main.yml

```diff
@@ -189,7 +189,7 @@
     kerberos_keytab_path: "{{kafka_broker_kerberos_keytab_path}}"
     kerberos_keytab_destination_path: "{{kafka_broker_keytab_path}}"
     kerberos_handler: "restart kafka"
-  when: "'GSSAPI' in kafka_broker_sasl_enabled_mechanisms or zookeeper_client_authentication_type == 'kerberos' or mds_broker_listener.sasl_protocol =='kerberos'"
+  when: "'GSSAPI' in kafka_sasl_enabled_mechanisms or zookeeper_client_authentication_type == 'kerberos' or mds_broker_listener.sasl_protocol =='kerberos'"
```

Comment:
is it possible that controller-controller communication happens over some protocol say kerberos and broker broker communication happens over some other protocol?
Also wanted to understand why changed this from kafka_broker_sasl to kafka_sasl

Reply:
kafka_sasl_enabled_mechanisms is list of all mechanisms enabled basically the union of kafka_broker_sasl_enabled_mechanisms and kafka_controller_sasl_enabled_mechanisms. This has been done for scenario where:
all:
   sasl_protocol: scram
   kafka_controller_sasl_protocol: kerberos
   
The above scenario represents that controller-controller and controller-broker communication needs to be done via kerberos so we need to configure kerberos, considering kafka_broker_sasl_enabled_mechanisms only we wouldn't be able to do that.